packages/block-editor/src/components/caption/index.native.js

```diff
@@ -1,65 +1,49 @@
 /**
  * External dependencies
  */
-import { View } from 'react-native';
+import { Platform } from 'react-native';
 
 /**
  * WordPress dependencies
  */
 import { RichText } from '@wordpress/block-editor';
-import { compose } from '@wordpress/compose';
-import { withDispatch, withSelect } from '@wordpress/data';
 import { __ } from '@wordpress/i18n';
 
-const Caption = ( { accessible, accessibilityLabel, onBlur, onChange, onFocus, isSelected, shouldDisplay, text } ) => (
-	<View style={ {	padding: 12, flex: 1, display: shouldDisplay ? 'flex' : 'none' } }
-		accessible={ accessible }
-		accessibilityLabel={ accessibilityLabel }
-		accessibilityRole={ 'button' }
-	>
-		<RichText
-			rootTagsToEliminate={ [ 'p' ] }
-			placeholder={ __( 'Write caption…' ) }
-			value={ text }
-			onChange={ onChange }
-			unstableOnFocus={ onFocus }
-			onBlur={ onBlur } // always assign onBlur as props
-			isSelected={ isSelected }
-			__unstableMobileNoFocusOnMount
-			fontSize={ 14 }
-			underlineColorAndroid="transparent"
-			textAlign={ 'center' }
-			tagName={ 'p' }
-		/>
-	</View>
-);
-
-export default compose( [
-	withSelect( ( select, { accessibilityLabelCreator, clientId } ) => {
-		const {
-			getBlockAttributes,
-			getSelectedBlockClientId,
-		} = select( 'core/block-editor' );
-		const { caption } = getBlockAttributes( clientId );
-		const accessibilityLabel = accessibilityLabelCreator ? accessibilityLabelCreator( caption ) : undefined;
-		const isBlockSelected = getSelectedBlockClientId() === clientId;
+// this platform difference is needed to avoid a regression described here:
+// https://github.com/WordPress/gutenberg/pull/18818#issuecomment-559818548
+const CAPTION_TAG_NAME = Platform.select( {
+	ios: 'p',
+	android: '',
+} );
```

Comment:
This was in the Gallery block to make it work correctly on Android due to the mentioned regression so I moved it to the Caption component itself since it was affecting the rest of Blocks using it.

Reply:
👍 Linking this up with the issue that was filed: https://github.com/wordpress-mobile/gutenberg-mobile/issues/1651

Reply:
I like that this platform divergence is consolidated into one place, OTOH, it seems we currently have to "choose" between two regressions for all captions: [Enter key not working](https://github.com/wordpress-mobile/gutenberg-mobile/issues/1651) or [First word not style-able](https://github.com/wordpress-mobile/gutenberg-mobile/issues/1674#issuecomment-565636271). This PR does not introduce these issues, but we do lose some granularity in how we can "choose which regression" :see_no_evil: for the different use cases. I'm not suggesting a change here, just noting, and hopefully those issues will be resolved, and this will soon be a moot point. :smile:

Reply:
Yeah you're right, by moving this to the Caption component it will affect Block level components with the [First word not style-able](https://github.com/wordpress-mobile/gutenberg-mobile/issues/1674#issuecomment-565636271) regression, but it will allow users to add new lines on Android... I wonder what users use the most, styling in captions or entering new lines 😅